crates/biome_analyze/src/lib.rs

```diff
@@ -842,33 +850,22 @@ impl<'analysis> AnalysisFilter<'analysis> {
             && self.enabled_rules.map_or(true, |enabled_rules| {
                 enabled_rules.iter().any(|filter| filter.match_group::<G>())
             })
-            && self.disabled_rules.map_or(true, |disabled_rules| {
-                !disabled_rules
-                    .iter()
-                    .any(|filter| filter.match_group::<G>())
-            })
+            && !self
+                .disabled_rules
+                .iter()
+                .any(|filter| matches!(filter, RuleFilter::Group(_)) && filter.match_group::<G>())
     }
 
     /// Return `true` if the rule `R` matches this filter
-    pub fn match_rule<R>(&self) -> bool
-    where
-        R: Rule,
-    {
-        self.match_group::<R::Group>()
+    pub fn match_rule<R: Rule>(&self) -> bool {
```

Comment:
nit: I am not sure the removal of `where` was necessary...

Reply:
This makes the code cleaner and more consistent with `match_group` (`match_group` doesn't use a `where` clause).